### PR TITLE
drivers/xbee: ported to netdev2

### PR DESCRIPTION
**Pull request by haukepetersen**
I think this is the last one that still uses `netdev1` and is without PR.

### CODE REVIEW

**Comment by haukepetersen** [comment]
tested with xbee+arduino-due

File: sys/include/net/gnrc/netdev2/xbee_adpt.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2016 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_gnrc
```
Comment (miri64) [inline sys/include/net/gnrc/netdev2/xbee_adpt.h]: `net_gnrc_netdev2` and `@addtogroup` if you do not define a new group.

File: sys/include/net/gnrc/netdev2/xbee_adpt.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * @{
+ *
+ * @file
+ * @brief       GNRC to XBee netdev2 glue code interface
```
Comment (miri64) [inline sys/include/net/gnrc/netdev2/xbee_adpt.h]: Maybe some reasoning why xbee needs its own glue code might be beneficial.

File: drivers/include/xbee.h [hunk 3]
```diff
@@ -171,7 +189,34 @@ extern const gnrc_netdev_driver_t xbee_driver;
...
+ *
+ * @return  the length of the created header in byte
+ * @return  -EOVERFLOW if @p payload_len is greater than XBEE_MAX_PAYLOAD_LENGTH
+ * @return  -ENOMSG if the given destination address has an invalid length
+ */
+int xbee_build_hdr(xbee_t *dev, uint8_t *xhdr, size_t payload_len,
+                   void *dst_addr, size_t addr_len);
+
+/**
+ * @brief   Extract the generic L2 header information from the XBee header
```
Comment (miri64) [inline drivers/include/xbee.h]: What kind of header is this?

Comment (haukepetersen) [inline drivers/include/xbee.h]: proprietary...

File: sys/include/net/gnrc/netdev2/xbee_adpt.h (hunk 2 above)
Comment (haukepetersen) [inline sys/include/net/gnrc/netdev2/xbee_adpt.h]: if you insist...

**Comment by haukepetersen** [comment]
fixed some documentation

File: sys/include/net/gnrc/netdev2/xbee_adpt.h [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * The XBee devices use their own proprietary format for communicating between
+ * the host CPU and the device over UART. The XBee device driver expects the
+ * data to send to be given in this format, hence we need to introduce an XBee
+ * specific adaption layer, that translates between GNRC and the XBee
+ * proprietary header format.
```
Comment (miri64) [inline sys/include/net/gnrc/netdev2/xbee_adpt.h]: Sorry for being nit-picky, but some reference to the IEEE 802.15.4 glue code is required here. The current text reads like xbee has nothing to do with IEEE 802.15.4 at all.

Comment (haukepetersen) [inline sys/include/net/gnrc/netdev2/xbee_adpt.h]: I doesn't w.r.t. to any 802.15.4 modules in RIOT... But I don't care, will add some infos.

**Comment by haukepetersen** [comment]
@miri64 added doc linking the adaption layer to 802.15.4. If you are still not happy, please propose a text that suits you and I will be happy to put it in :-)

**Comment by miri64** [comment]
Apart from language I'm good for now :+1:.

File: drivers/xbee/include/xbee_params.h [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+#ifndef XBEE_PARAMS
+#define XBEE_PARAMS             { .uart      = XBEE_PARAM_UART, \
+                                  .br        = XBEE_PARAM_BR, \
+                                  .pin_sleep = XBEE_PARAM_PIN_SLEEP, \
+                                  .pin_reset = XBEE_PARAM_PIN_RESET }
+#endif
+/**@}*/
+
+/**
+ * @brief   AT86RF231 configuration
```
Comment (Yonezawa-T2) [inline drivers/xbee/include/xbee_params.h]: This should be XBee.

Comment (haukepetersen) [inline drivers/xbee/include/xbee_params.h]: fixed

**Comment by kYc0o** [comment]
OK everything looks ok, I tested on arduino-due as well with success on tests/driver_xbee and examples/gnrc_minimal. However, can you just modify the Makefile on this last example?

**Comment by haukepetersen** [comment]
> However, can you just modify the Makefile on this last example?

Which Makefile, and modify to what?

**Comment by kYc0o** [comment]
The Makefile in examples/gnrc_minimal, on which we give instructions on how enable the use of XBee for this example. These instructions are intended for the previous driver's style.

**Comment by haukepetersen** [comment]
I see, will fix

**Comment by haukepetersen** [comment]
done, also squashed and rebased.

**Pull request merged**